packages/embed/src/apple-pay/apple-pay.ts

```diff
@@ -1,11 +1,70 @@
 import { SubjectManager } from 'subjects'
+import { error, log, warn } from 'utils'
+
+export const APPLE_PAY_TIMEOUT = 1000
 
 declare global {
   interface Window {
     ApplePaySession?: ApplePaySession
   }
 }
 
+export const loadApplePaySdk = () => {
+  return Promise.race([
+    new Promise<boolean>((resolve) => {
+      if (typeof window === 'undefined') {
+        return resolve(false)
+      }
+
+      if (window.ApplePaySession || document.getElementById('apple-pay-sdk')) {
```

Comment:
What's the reason for the `getElementById` check here? Is it possible for that to exist but `window.ApplePaySession` to not exist?

Reply:
It's to avoid continuing if there's already a script with that id on the page (the one we add). That could happen in case, say, a React route / page re-renders quickly and the function is called again (but we're already adding the script)